sys/sys/types.h

```diff
@@ -300,6 +300,11 @@ typedef	__uint64_t	uoff_t;
 typedef	char		vm_memattr_t;	/* memory attribute codes */
 typedef	struct vm_page	*vm_page_t;
 
+typedef	int32_t		intptr32_t;
+typedef	uint32_t	uintptr32_t;
+typedef	int64_t		intptr64_t;
+typedef	uint64_t	uintptr64_t;
```

Comment:
It's a bit weird to have this on ILP32, and a bit weird for these to exist outside of compat code

Reply:
I originally wanted to put these in <compat/freebsdNN/freebsdNN.h>, but many places include freebsdNN_proto.h without freebsdNN.h and so failed to compile.  One of those places is the generated freebsdNN_sysent.c, so I tried to find a header that was already included by various places.  Given that it is kernel only I could guard the typedefs with #ifdef COMPAT_FREEBSDnn.